bgpd/bgp_mplsvpn.c

```diff
@@ -3583,3 +3583,37 @@ void bgp_vpn_leak_export(struct bgp *from_bgp)
 		}
 	}
 }
+
+mpls_label_t bgp_mplsvpn_adv_label(struct bgp_path_info *pi, struct peer *to,
+				   afi_t afi, safi_t safi)
+{
+	struct peer *from;
+	mpls_label_t remote_label;
+	bool reflect;
+
+	if (!pi || !to || !to->bgp)
+		return MPLS_INVALID_LABEL;
+
+	remote_label = pi->extra ? pi->extra->label[0] : MPLS_INVALID_LABEL;
+	from = pi->peer;
+
+	if (pi->sub_type == BGP_ROUTE_IMPORTED || from == NULL)
+		/* local or imported mpls-vpn route
+		 * we use the available label
+		 */
+		return remote_label;
+
+	reflect = (from != NULL) && ((from->sort == BGP_PEER_IBGP) &&
+				     (to->sort == BGP_PEER_IBGP));
+	if (reflect &&
```

Comment:
No need for the `reflect` variable because you use it only once.

Reply:
done